src/main/java/com/google/javascript/gents/TypeConversionPass.java

```diff
@@ -647,8 +647,9 @@ static ClassMemberDeclaration newDeclaration(Node n, Map<String, Node> classes)
     @Nullable
     static ClassMemberDeclaration newDeclarationOnThis(Node n) {
       Node fullName = getFullName(n);
-      // Node MUST start with "this."
-      if (!fullName.isGetProp() || !containsThis(fullName)) {
+      // Node MUST start with "this." and be shallow, i.e. "this.foo".
+      // "this.foo.bar" and other nestings are not declarations and are ignored.
+      if (!fullName.isGetProp() || !fullName.getFirstChild().isThis()) {
```

Comment:
Can you formulate the comment such that it's more obvious that if `getFirstChild().isThis()` is true, this can only be a single `this.x` step because `fullName` is a binary tree, i.e. there cannot be a third and fourth child?